trunk/SUAVE/Methods/Propulsion/electric_motor_sizing.py

```diff
@@ -120,17 +120,38 @@ def compute_optimal_motor_parameters(motor,prop):
     io                   = motor.no_load_current
     v                    = motor.nominal_voltage 
     omeg                 = prop.angular_velocity
-    etam                 = motor.efficiency
-    start_kv             = 1
-    end_kv               = 15     
-    possible_kv_vals     = np.linspace(start_kv,end_kv,(end_kv-start_kv)*20 +1 , endpoint = True) * Units.rpm
-    res_kv_vals          = ((v-omeg/possible_kv_vals)*(1.-etam*v*possible_kv_vals/omeg))/io  
-    positive_res_vals    = np.extract(res_kv_vals > 0 ,res_kv_vals) 
-    kv_idx               = np.where(res_kv_vals == min(positive_res_vals))[0][0]   
-    kv                   = possible_kv_vals[kv_idx]  
-    res                  = min(positive_res_vals) 
-    
-    motor.speed_constant = kv
-    motor.resistance     = res 
+    etam                 = motor.efficiency 
+    motor.speed_constant = optimize_kv(io, v , omeg,  etam)
+    motor.resistance     = ((v-omeg/motor.speed_constant)*(1.-etam*v*motor.speed_constant/omeg))/io
+    
+    return motor
+
+def optimize_kv(io, v , omeg,  etam): 
+    ''' Optimizer for compute_optimal_motor_parameters function  
+    
+    Source:
+    N/A
+    
+    Inputs:
+    motor    (to be modified)
+    
+    Outputs:
+    motor.
+      speed_constant     [untiless]
+      no_load_current    [amps]
+    '''        
+    # objective 
+    objective = lambda x: ((v-omeg/x[0])*(1.-etam*v*x[0]/omeg))/io
+    
+    # bounds 
+    bnds = [(0,100)]
```

Comment:
add this as a defaulted parameter. I assume these bounds are the Kv. 100 Kv is low for a multicopter

Reply:
default parameter added